string affordance
bool add
---
bool success
